deps/rabbit/src/rabbit_channel_tracking.erl

```diff
@@ -116,22 +115,46 @@ handle_cast({user_deleted, Details}) ->
             delete_tracked_channel_user_entry, [Username]),
     ok;
 handle_cast({node_deleted, Details}) ->
-    Node = pget(node, Details),
-    rabbit_log_channel:info(
-        "Node '~s' was removed from the cluster, deleting"
-        " its channel tracking tables...", [Node]),
-    delete_tracked_channels_table_for_node(Node),
-    delete_per_user_tracked_channels_table_for_node(Node).
+    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
+        true ->
+            ok;
+        false ->
+            Node = pget(node, Details),
+            rabbit_log_channel:info(
+              "Node '~s' was removed from the cluster, deleting"
+              " its channel tracking tables...", [Node]),
+            delete_tracked_channels_table_for_node(Node),
+            delete_per_user_tracked_channels_table_for_node(Node)
+    end.
 
 -spec register_tracked(rabbit_types:tracked_channel()) -> ok.
 -dialyzer([{nowarn_function, [register_tracked/1]}]).
 
-register_tracked(TrackedCh =
-  #tracked_channel{node = Node, name = Name, username = Username}) ->
+register_tracked(TrackedCh = #tracked_channel{node = Node}) when Node == node() ->
+    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
+        true -> register_tracked_ets(TrackedCh);
+        false -> register_tracked_mnesia(TrackedCh)
+    end;
+register_tracked(#tracked_channel{node = Node} = TrackedCh) ->
+    rabbit_misc:rpc_call(Node, ?MODULE, register_tracked, [TrackedCh]).
```

Comment:
You're right. Even though it's exported, as it's a callback implementation, it's only used internally. I just did a search on the whole server repo and isn't used anywhere else. The function call is guarded both in `channel_created` and `channel_closed` as well as it's connection equivalents. I'll remove the rpc call as it will never be used